lib/galaxy/datatypes/anvio.py

```diff
@@ -36,17 +36,18 @@ def generate_primary_file(self, dataset=None):
                 if composite_file.optional:
                     opt_text = " (optional)"
                 missing_text = ""
-                if not os.path.exists(os.path.join(dataset.extra_files_path, composite_name)):
+                if dataset and not os.path.exists(os.path.join(dataset.extra_files_path, composite_name)):
```

Comment:
The method is inherited from the base class: AnvioComposite < Html < Text < Data. We can't remove the default in Data since it is indeed optional in many of its subclasses; but if we remove it only here, mypy will complain after type hints are added (can't break the signature of a parent class method). 
Another option is to add an `assert dataset` statement, or a mypy ignore comment; I think explicitly checking the value is the easiest and safest option here, but we could use either one instead.

Reply:
Yeah, but all the calling paths provide dataset afaict, what's the point then to have it optional and check for cases that can never happen?